File: 1461-count-all-valid-pickup-and-delivery-options/count-all-valid-pickup-and-delivery-options.py
class Solution:
    def countOrders(self, n: int) -> int:
        mod=1e9+7
        if n==1:
            return 1
        result=1
        for i in range(2,n+1):
            space=(i-1)*2+1
            temp=space*(space+1)//2
            result*=temp
            result%=mod
        return int(result)

File: 1461-count-all-valid-pickup-and-delivery-options/count-all-valid-pickup-and-delivery-options.java
class Solution {
    public int countOrders(int n) {
        int mod=1000000007;
        if (n==1)return 1;
        long result=1;
        for(int i=2;i<=n;i++){
            int s=(i-1)*2+1;
            int temp=s*(s+1)/2;
            result*=temp;
            result%=mod;
        }
        return (int)result;
    }
}